client/src/components/History/CurrentHistory/HistoryOperations/Operations.vue

```diff
@@ -52,19 +52,19 @@ export default {
         onCopy() {
             iframeRedirect("/dataset/copy_datasets");
         },
-        async unhideAll() {
-            await this.runOperation(() => unhideAllHiddenContent(this.history));
+        unhideAll() {
+            this.runOperation(() => unhideAllHiddenContent(this.history));
         },
-        async deleteAllHidden() {
-            await this.runOperation(() => deleteAllHiddenContent(this.history));
+        deleteAllHidden() {
+            this.runOperation(() => deleteAllHiddenContent(this.history));
         },
-        async purgeAllDeleted() {
-            await this.runOperation(() => purgeAllDeletedContent(this.history));
```

Comment:
Can you give a bit of rationale for the change here?  Looking at the calls involved the `async` decorator is appropriate for any of these operations.

Reply:
Thank you, that is a good question. So `purgeAllDelted` is called as part of [`@ok="purgeAllDeleted"`](https://github.com/galaxyproject/galaxy/blob/dev/client/src/components/History/CurrentHistory/HistoryOperations/Operations.vue#L36). This is a button click event which does not have to wait for the result of `purgeAllDeleted` when calling it. So adding async/await here has no effect. Also `purgeAllDelted` is just waiting for a single call to complete. In this case just returning the promise or async function would preserves the promise chain if needed.

Reply:
xref: https://github.com/galaxyproject/galaxy/pull/13881/files#diff-9f0b0b19054c241dfd008b38f14c385df1073ecef759a3e778d7ffe136e1d2e5R87

Reply:
Nice topic! 
For me, propagating the `async` makes it more clear that this function returns a promise and *can* be awaited (even if it doesn't have to like in the button click event) without having to dig down the call until I found the promise. Also, I'm a bit influenced by my past in C# :laughing:  but yeah, I agree this is more verbose and is not really part of a public interface so... :+1:

Reply:
> For me, propagating the async makes it more clear that this function returns a promise and can be awaited (even if it doesn't have to like in the button click event) without having to dig down the call until I found the promise.

Yep, exactly this, it makes things clearer and easier to read without digging through the full function chain, for me.  And this particular component example isn't part of a public API, but https://github.com/galaxyproject/galaxy/pull/13890/files#diff-75b396c25a0e1a2d582107df23dd4c97656c91cc59bb9331cf646d2da8c47ec3 is a distinct interface where that'd be best left in IMO.

Reply:
A related question appeared at https://stackoverflow.com/questions/35201881/should-i-use-async-await-for-every-method-that-returns-a-task: The best response states: `Realistically you should only really use the await keyword when there is other code in the method that needs to be run after the code that returns the task. Because that code needs to await for it to complete before it runs.`

Reply:
Thanks @guerler for the research, that is interesting indeed! I guess my concern for not knowing that these functions are actually returning a promise can be fixed more cleanly by using a better naming convention.

Maybe is a good idea to add an `Async` postfix to all these kinds of functions? Like, in this case, `purgeAllDeletedAsync` ? then we don't need to explicitly add any *async/await* syntax while being aware of the promise involved.

Reply:
The second thing @guerler posted is about C#, not sure if that applies. I can't judge the merit of the async keyword, but I'd naively think that the best way to achieve

> makes it more clear that this function returns a promise and can be awaited

is to have type annotations. Can we add these without a full shift to typescript ? Seems like JSDoc comments can double as type annotations ?

Reply:
Agree, if we can use JSDoc for that it is even better than the naming convention :+1: 

Although I would love to switch to TS soonish :stuck_out_tongue: